### PR TITLE
Implement match game count warning check

### PR DESCRIPTION
**Pull request by myssto**
Depends on #512
Part of #479

### CODE REVIEW

File: DataWorkerService.Tests/AutomationChecks/Matches/MatchGameCountCheckTests.cs [hunk 1]
```diff
@@ -107,4 +107,45 @@ MatchRejectionReason expectedRejectionReason
...
+    public void Check_Passes_WhenApplyingWarningFlag()
+    {
+        // Arrange
+        Match match = SeededMatch.Generate(warningFlags: MatchWarningFlags.None);
+
+        SeededGame.Generate(verificationStatus: VerificationStatus.Verified, match: match);
+        SeededGame.Generate(verificationStatus: VerificationStatus.Verified, match: match);
+        SeededGame.Generate(verificationStatus: VerificationStatus.Verified, match: match);
+        SeededGame.Generate(verificationStatus: VerificationStatus.Verified, match: match);
+
```
Comment (hburn7) [inline DataWorkerService.Tests/AutomationChecks/Matches/MatchGameCountCheckTests.cs:121]: for loop me thinks

**Review by hburn7 — CHANGES_REQUESTED** [review]

File: DataWorkerService.Tests/AutomationChecks/Matches/MatchGameCountCheckTests.cs (hunk 1 above)
Comment (myssto) [reply DataWorkerService.Tests/AutomationChecks/Matches/MatchGameCountCheckTests.cs:121]: methinks it will take the same amount of space :P

Comment (hburn7) [reply DataWorkerService.Tests/AutomationChecks/Matches/MatchGameCountCheckTests.cs:121]: true wisdom lies not in the space it takes, but in the loops that spare us the tedium of repetition!

Comment (myssto) [reply DataWorkerService.Tests/AutomationChecks/Matches/MatchGameCountCheckTests.cs:121]: me when I wrote the one line and then alt + shift + down arrow 3 times in a row

**Review by hburn7 — APPROVED** [review]

**Pull request merged**